File: python/test/api/arch/role_demo_test.py
import src.api.arch.role_demo as role_demo
from src.model.role.role_list_query_param import RoleListQueryParam



# 测试 列出角色
def roleList():
    role_list_query_param = RoleListQueryParam(0, 10)
    result = role_demo.roleList(role_list_query_param)
    print('roleList result:', result)


# 测试 创建角色
def roleCreate():
    result = role_demo.roleCreate('python-name',2300)
    print('roleCreate result:', result)

# 测试 修改角色
def roleUpdate():
    result = role_demo.roleUpdate('python-name-1',2319,2329)
    print('roleUpdate result:', result)



# 测试 删除角色
def roleDelete():
    result = role_demo.roleDelete(2333)
    print('roleDelete result:', result)


# 测试 列出角色下的所有成员
def roleMemberList():
    result = role_demo.roleMemberList(2334,0,10)
    print('roleMemberList result:', result)


# 测试 批量给已有的成员设置自建角色
def roleAddMembers():
    result = role_demo.roleAddMembers(2334,['R-TygTJmwo','R-RS8aHUlr'])
    print('roleAddMembers result:', result)

#
# 测试 为自建角色批量移除成员
def roleRmoveMembers():
    result = role_demo.roleRmoveMembers(2334, ['R-TygTJmwo', 'R-RS8aHUlr'])
    print('roleRmoveMembers result:', result)


if __name__ == '__main__':
    roleCreate()
    roleList()
    roleUpdate()
    roleDelete()
    roleMemberList()
    roleAddMembers()
    roleRmoveMembers()

File: java/src/test/java/api/arch/RoleDemoTest.java
package api.arch;

import model.role.RoleListQueryParam;

import java.util.Arrays;
import java.util.Map;

public class RoleDemoTest {

    public static void main(String[] args) throws Exception {
        RoleDemo roleDemo = new RoleDemo();
        // 列出角色
        roleList(roleDemo);
        // 创建角色
        roleCreate(roleDemo);
        // 更新角色
        roleUpdate(roleDemo);
        // 删除角色
        roleDelete(roleDemo);
        // 列出角色下的所有成员
        roleMemberList(roleDemo);
        // 批量给已有的成员设置角色
        roleAddMembers(roleDemo);
        // 为自建角色批量移除成员
        roleRmoveMembers(roleDemo);
    }

    private static void roleRmoveMembers(RoleDemo roleDemo) throws Exception {
        Map<String, Object> result = roleDemo.roleRmoveMembers(2304, Arrays.asList("importUserName2","userName2"));
        System.out.println("roleRmoveMembers result\n" + result);
    }

    private static void roleAddMembers(RoleDemo roleDemo) throws Exception {
        Map<String, Object> result = roleDemo.roleAddMembers(2304, Arrays.asList("importUserName2","userName2"));
        System.out.println("roleAddMembers result\n" + result);
    }

    private static void roleMemberList(RoleDemo roleDemo) throws Exception {
        Map<String, Object> result = roleDemo.roleMemberList(2304,0,100);
        System.out.println("roleMemberList result\n" + result);
    }

    private static void roleDelete(RoleDemo roleDemo) throws Exception {
        Map<String, Object> result = roleDemo.roleDelete(2310);
        System.out.println("roleDelete result\n" + result);
    }

    private static void roleUpdate(RoleDemo roleDemo) throws Exception {
        Map<String, Object> result = roleDemo.roleUpdate("JavaName2",2302,2326);
        System.out.println("roleUpdate result\n" + result);
    }

    private static void roleCreate(RoleDemo roleDemo) throws Exception {
        Map<String, Object> result = roleDemo.roleCreate("JavaName1",2300);
        System.out.println("roleCreate result\n" + result);
    }

    private static void roleList(RoleDemo roleDemo) throws Exception {
        RoleListQueryParam queryParam = new RoleListQueryParam();
        queryParam.setSkip(0);
        queryParam.setLimit(10);
        queryParam.setHas_internal(true);
        queryParam.setHas_sync(true);
        Map<String, Object> result = roleDemo.roleList(queryParam);
        System.out.println("roleList result\n" + result);
    }
}
